### DISCUSSION TITLE
enhancement, apply the same method as with MQTT to obtain a custom-variant of SPOT-table

### DISCUSSION THREAD

**Discussion by silliewimons** [SBFspot/SBFspot#510] [Ideas]
under the restriction that one might only choice:  either -sp1  either -csp1  either -MQTT  
the new option   -csp1   will generate custom-spot table       (and is at least exclusive towards -sp1)
then the code and system-approach that is used to generate MQTT output mentioned as a (csv-format) 
can also be used to obtain a customised format of the SPOT-table,  (format defined in the config-file)
!!! if 100% compatible table format is wanted then following modification are needed in keywords (extract config for v390)
```
PACTot			GridMsTotW		Power (AC)  
# PDC1/PDC2			DcMsWatt		DC power input string 1/2
# PDCoth						DC power input string 3 & higher                     <--separated from PDC1/PDC2     
# UDC1/UDC2			DcMsVol			DC voltage input string 1/2
# UDCoth						DC voltage input strings 3 and higher       <--separated from 
# IDC1/IDC2			DcMsAmp			DC current input string 1/2
# IDCoth						DC current input strings 3 and higher                  <--separate from IDC1/IDC2
# PDCTot 			DcMsWatt 		DC power all strings (calc)
     . . . . . . 
```
other fields/keywords that can be usefull as part of the custom-table
  Efficiency   (calculated)  from  PACTot / PDCTot
  EMPTY       for compatibility reasons - eg  when a field is in use with the other outputs, 
  TimeZone  ?? might be usefull with battery operated systems  (my solution is to NOT use DST at all) 
??whether a keyword  can be allowed more then once ??

background, with the addition of multi-string support the format of 'SPOT-table' is different from one inverter to another
eg. when a classical 2strings-inverter  AND  a multistring inverter  is monitored in one installation
      then one  must be carefull in comparing the results,

kr wim

**Comment by sbf-** [comment]
If you want a custom CSV file export, I propose to use SQLite. MQTT or yet another export format is not the right way to go...
In SQLite you can create one or more views with the fields you want to export:

```
C:\Program Files (x86)\SQLite>sqlite3 d:\Public\SMAdata\SBFspot.db
SQLite version 3.15.2 2016-11-28 19:13:37
Enter ".help" for usage hints.
sqlite> .headers on
sqlite> .mode csv
sqlite> .separator ";"
sqlite> .output c:\\temp\\spot.csv
sqlite> select * from vwspotdata where serial=2100123456;
sqlite> .quit
```
For more information: https://www.sqlitetutorial.net/sqlite-tutorial/sqlite-export-csv/

**Reply by silliewimons** [reply]
@sbf,  thank you for the information and the proposal,   in view of multi-inverter  &  multi-installation,
is the  'plant-name'   part of the config.file available as entry for database??

kr wim

**Comment by silliewimons** [comment]
using the MQTT options, it is possible too simulate the equivalent of the SmartScreen output,
- altough some limitations needed a workaround 

kr  sillie
